node/node.go

```diff
@@ -724,10 +729,23 @@ func (node *Node) handleNewChainHeads(ctx context.Context, prevHead types.TipSet
 			prevHead = newHead
 
 			if node.StorageMiner != nil {
-				err := node.StorageMiner.OnNewHeaviestTipSet(newHead)
-				if err != nil {
+				if err := node.StorageMiner.OnNewHeaviestTipSet(newHead); err != nil {
 					log.Error(err)
 				}
+				if node.StorageFaultMonitor != nil {
+					height, err := newHead.Height()
```

Comment:
BLOCKING: This is too much logic to put in node. It's also untested. At the very least, StorageFaultMiner should have its own (unit tested) `OnNewHeaviestTipSet` metthod. Since fault handling it intimately tied to storage mining, it might be better for the SlashMonitor to be constructed withing the StorageMiner and have it trigger slashing from its `OnNewHeaviestTipSet` method.

Reply:
good call, this is from the weeks-old original invocation and the design changed a lot while you were gone.  it went unexamined, and I agree that's a better idea.